src/App.tsx

```diff
@@ -1,61 +1,154 @@
+import { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number | null) {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todosWithUsers: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(todosWithUsers);
+
+  const [selectedUserId, setSelectedUserId] = useState<number>(0);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isUserSelectError, setIsUserSelectError] = useState(false);
+  const [isTitleError, setIsTitleError] = useState(false);
+
+  const clearForm = () => {
+    setSelectedUserId(0);
+    setNewTodoTitle('');
+    setIsTitleError(false);
+    setIsUserSelectError(false);
+  };
+
+  const getNewId = (arr: { id: number }[]): number => {
+    const ids = arr.map((item) => item.id);
+    const max = Math.max(...ids);
+
+    return Number.isFinite(max)
+      ? max + 1
+      : 1;
+  };
+
+  const handleSetTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodoTitle(event.target.value);
+    setIsTitleError(false);
+  };
+
+  const handleSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(+event.target.value);
+    setIsUserSelectError(false);
+  };
+
+  const handleSubmit = ((event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const validateNewTitle = newTodoTitle.trim();
+
+    if (!selectedUserId || !validateNewTitle) {
+      setIsTitleError(!validateNewTitle);
+      setIsUserSelectError(!selectedUserId);
+
+      return;
+    }
+
+    setTodos((prevTodos) => {
+      const newTodo: Todo = {
+        id: getNewId(prevTodos),
+        title: validateNewTitle,
+        userId: selectedUserId,
+        completed: false,
+        user: getUser(selectedUserId),
+      };
+
+      return [...prevTodos, newTodo];
+    });
+
+    clearForm();
+  });
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/todos"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="taskTitle">
+            {'Title: '}
```

Comment:
please don't add the spaces this way, you can use the margins and padding instead